### PR TITLE
Refacto/native resources

### PR DESCRIPTION
**Pull request by l0r1s**

### CODE REVIEW

**Comment by l0r1s** [comment]
Tests to be added

File: javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+  constructor(
+    client: Client,
+    protected readonly namespace: string,
+    protected readonly nodeSetupConfig: Node,
+  ) {
+    const nodeName = nodeSetupConfig.name;
+    this.configPath = `${client.tmpDir}/${nodeName}/cfg`;
+    this.dataPath = `${client.tmpDir}/${nodeName}/data`;
+    this.relayDataPath = `${client.tmpDir}/${nodeName}/relay-data`;
```
Comment (pepoviola) [inline javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts:26]:
```suggestion
    const nodeRootPath = `${client.tmpDir}/${nodeSetupConfig.name}`;
    this.configPath = `${nodeRootPath}/cfg`;
    this.dataPath = `${nodeRootPath}/data`;
    this.relayDataPath = `${nodeRootPath}/relay-data`;
```

**Review by pepoviola — COMMENTED** [review]
Looks good! A couple of styling suggestion/questions.
Thanks!!

File: javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  private getZombieRole() {
+    const { zombieRole, validator } = this.nodeSetupConfig;
+
+    if (zombieRole) {
+      return zombieRole;
+    } else if (validator) {
+      return "authority";
+    }
+
+    return "full-node";
```
[the remark this answers is not in the document]
Comment (l0r1s) [reply javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts:106]: One ternary is ok, two ternary is not readable. Eventually we can do this:
```suggestion
    if (zombieRole) {
      return zombieRole;
    } 

    return validator ? "authority" : "full-node";
```

File: javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      return memo;
+    }, {} as ProcessEnvironment);
+  }
+
+  protected generateNodeSpec(
+    ports: any[],
+    command: string[],
+    zombieRole: string,
+    env: ProcessEnvironment,
+  ): any {
```
Comment (wirednkod) [inline javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts:123]: Can we avoid `any` as a type? This is one of the pains we have at the moment :)

File: javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (wirednkod) [reply javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts:106]: or just:
```suggestion
    if (zombieRole) return zombieRole;

    return validator ? "authority" : "full-node";
```

File: javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    protected readonly nodeSetupConfig: Node,
+  ) {
+    const nodeName = nodeSetupConfig.name;
+    this.configPath = `${client.tmpDir}/${nodeName}/cfg`;
+    this.dataPath = `${client.tmpDir}/${nodeName}/data`;
+    this.relayDataPath = `${client.tmpDir}/${nodeName}/relay-data`;
+  }
+
+  public async generateSpec() {
+    await this.createDirectories();
```
Comment (wirednkod) [inline javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts:30]: Maybe add a try-catch that will throw a "generation failed" error?

File: javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (l0r1s) [reply javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts]: Done

File: javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    const { env } = this.nodeSetupConfig;
+
+    return env.reduce((memo, item: envVars) => {
+      memo[item.name] = item.value;
+      return memo;
+    }, {} as ProcessEnvironment);
+  }
+
+  protected generateNodeSpec(
+    ports: any[],
```
[the remark this answers is not in the document]
Comment (l0r1s) [reply javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts]: Done

File: javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts (hunk 3 above)
Comment (l0r1s) [reply javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts]: Done

File: javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts (hunk 1 above)
Comment (l0r1s) [reply javascript/packages/orchestrator/src/providers/native/resources/node.resource.ts]: Done

**Review by wirednkod — APPROVED** [review]

**Review by pepoviola — APPROVED** [review]

**Pull request merged**